kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration-audit-replication/src/test/java/org/kie/server/spring/boot/autoconfiguration/audit/replication/AuditDataReplicationKieServerTest.java

```diff
@@ -0,0 +1,258 @@
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.services.api.RuntimeDataService;
+import org.jbpm.services.api.UserTaskService;
+import org.jbpm.services.task.audit.impl.model.AuditTaskImpl;
+import org.jbpm.services.task.audit.impl.model.BAMTaskSummaryImpl;
+import org.jbpm.services.task.audit.impl.model.TaskEventImpl;
+import org.jbpm.services.task.audit.impl.model.TaskVariableImpl;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.runtime.manager.audit.ProcessInstanceLog;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.KieServerConfigItem;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.services.api.KieServer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK, classes = ApplicationSender.class)
+@TestPropertySource(locations = "classpath:application-integrationtest.properties")
+public class AuditDataReplicationKieServerTest {
+
+    private static final Long TIMEOUT = 10000L;
+    private static final String USER_GENERIC = "salaboy";
+    private static final String USER_NOMINATED = "krisv";
+    private static final String USER_ADMIN = "Administrator";
+
+    @Autowired
+    private EntityManagerFactory originalEntityManagerFactory;
+
+    @Autowired
+    @Qualifier("auditEntityManagerFactory")
+    private EntityManagerFactory auditEntityManagerFactory;
+
+    @Autowired
+    private DataSource datasourceOriginal;
+
+    @Autowired
+    @Qualifier("datasource-replica")
+    private DataSource datasourceReplica;
+
+    @Autowired
+    private KieServer kieServer;
+
+    @Autowired
+    private ProcessService processService;
+
+    @Autowired
+    private RuntimeDataService runtimeDataService;
+
+    @Autowired
+    private UserTaskService userTaskService;
+
+    protected static final EmbeddedActiveMQ embedded = new EmbeddedActiveMQ();
+
+    @Autowired
+    @Qualifier("auditDataReplicationQueueConsumer")
+    private  AuditDataReplicationJMSQueueConsumer consumer;
+
+    @BeforeClass
+    public static void startUp() throws Exception {
+        KieJarBuildHelper.createKieJar("src/test/resources/kjar/");
+        Configuration config = new ConfigurationImpl();
+        config.setSecurityEnabled(false);
+        config.addAcceptorConfiguration("amqp-acceptor", "tcp://localhost:10022?protocols=AMQP");
+        embedded.setConfiguration(config);
+        embedded.start();
+        
+    }
+
+    @AfterClass
+    public static void shutDown() throws Exception {
+        embedded.stop();
```

Comment:
It might be worth removing `data` and `transactions-logs` folders after test case execution. It will fail next test otherwise.

Reply:
in post integration test pom.xml